src/BccCode.Linq/QueryProvider/ApiQueryProvider.cs

```diff
@@ -739,10 +761,105 @@ protected override Expression VisitMethodCall(MethodCallExpression node)
                         _where.Append("}}");
 
                         return Expression.Empty();
+                    case nameof(object.ToString):
+                    {
+                        if (node.Arguments.Count != 0)
+                        {
+                            throw new NotSupportedException(
+                                "object.ToString method call is currently only supported in a Where clause without arguments");
+                        }
+
+                        Debug.Assert(node.Object != null);
+                        var objectType = node.Object.Type;
+                        if (objectType.IsGenericType && objectType.GetGenericTypeDefinition() == typeof(Nullable<>))
+                        {
+                            objectType = Nullable.GetUnderlyingType(objectType);
+                            Debug.Assert(objectType != null);
+                        }
+
+                        if (TypeHelper.IsNumberType(objectType))
+                        {
+                            throw new NotSupportedException(
+                                "object.ToString on number types (int, long, short, ...) is not supported in a Where clause");
+                        }
+                        else if (objectType == typeof(bool))
+                        {
+                            throw new NotSupportedException(
+                                $"ToString on type {objectType.FullName} is not supported in a Where clause");
+                        }
+
+                        if (objectType == typeof(string) || objectType == typeof(Guid) || objectType == typeof(DateTime)
+#if NET6_0_OR_GREATER
+                                                                 || objectType == typeof(DateOnly)
+#endif
+                           )
+                        {
+                            // Officially supported types with ToString()
+                        }
+                        else
+                        {
+                            Debug.WriteLine($"{nameof(ApiQueryProvider)}: Using object.ToString() call on type {objectType.FullName} is not officially tested in a Where clause. You might end up getting unexpected results.");
+                        }
+
+                        // We skipp interpreting ToString() because it will already be passed as string in JSON
+
+                        var obj = Visit(node.Object);
+                        Debug.Assert(obj != null);
+                        Debug.Assert(obj.NodeType == ExpressionType.Default);
+                        return Expression.Empty();
+                    }
+                        break;
                     default:
                         throw new NotSupportedException($"Not supported method call: {node.Method.DeclaringType?.FullName}.{node.Method.Name}");
                 }
             }
+            else if (node.Method.Name == "ToString")
+            {
+                if (node.Arguments.Count != 0)
```

Comment:
Is the block of code below more or less a duplicate of the code above?

Reply:
No, I think this is the most important code point where we will end up most of the times:
The upper implementation is only used when you call the default `ToString` method of a class which is not overloaded. This method will be called when you have `ToString` called on structs and classes which overload the `ToString` method.

Reply:
I put the code now in a single function. We need it at both places in code: when using nullable struct types, the `object.ToString` switch case runs, and when using structs without nullable, the `node.Method.Name == "ToString"` implementation runs.

Reply:
This was a good point!